### PR TITLE
Collect code coverage of tsbench benchmarks

### PR DESCRIPTION
**Pull request by akuzm**
This will allow us to find missing benchmarks.

Runs on main + on schedule daily, takes half an hour. Also has assertions enabled to increase test coverage.

Disable-check: force-changelog-file

### CODE REVIEW

**Comment by jnidzwetzki** [comment]
@akuzm How can these reports be accessed in general? Just as an action artifact?

File: .github/workflows/tsbench-coverage.yaml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+"on":
+  schedule:
+    - cron: '0 3 * * *'
+  push:
+    branches:
+      - main
+      - trigger/tsb-coverage
+  pull_request:
+    paths: .github/workflows/tsbench-coverage.yaml
+
```
Comment (jnidzwetzki) [inline .github/workflows/tsbench-coverage.yaml:13]: Is there a specific reason why we don't allow `workflow_dispatch` here? Might be useful to be able to trigger the workflow directly from the GitHub UI.

File: .github/workflows/tsbench-coverage.yaml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+          psql -c "create extension timescaledb;"
+
+          cd tsbench
+          source .venv/bin/activate
+          tsbench-deployer --verbose --cache-s3-datasets
+          LANG=C LC_NUMERIC=C tsbench --max-executions=1 --verbose --verbose \
+            --with-connection pgsq://localhost:5432/postgres
+
+          # Postgres should survive.
+          psql -c "select 1;"
```
Comment (jnidzwetzki) [inline .github/workflows/tsbench-coverage.yaml:156]: What is the reason for this test? Just to make sure PostgreSQL did not crash during the benchmark execution?

File: .github/workflows/tsbench-coverage.yaml [hunk 3]
```diff
@@ -0,0 +1,204 @@
+# A job to inspect code coverage of current TSBench benchmarks, and also run
+# them with assertions enabled to have more test coverage.
+name: TSBench Code Coverage
+"on":
+  schedule:
+    - cron: '0 3 * * *'
```
Comment (jnidzwetzki) [inline .github/workflows/tsbench-coverage.yaml:6]: Do we need to schedule this action? Shouldn't this scheduled run produce the same output as the last run on `main`?

**Comment by akuzm** [comment]
> @akuzm How can these reports be accessed in general? Just as an action artifact?

Yes, only as artifact. I'm not sure if it's possible to add a separate configuration for codecov.

File: .github/workflows/tsbench-coverage.yaml (hunk 2 above)
Comment (akuzm) [reply .github/workflows/tsbench-coverage.yaml:156]: Yes.

**Comment by akuzm** [comment]
Maybe this action should go to tsbench repo instead...

**Comment by akuzm** [comment]
I implemented this in tsbench repository.

**Pull request closed without merging**